src/solvers/imex_tableaus.jl

```diff
@@ -19,13 +19,29 @@ default values for `c_exp` and `c_imp` assume that it is internally consistent.
 The explicit tableau must be strictly lower triangular, and the implicit tableau
 must be lower triangular (only DIRK algorithms are currently supported).
 """
-struct IMEXTableau{VS <: StaticArrays.StaticArray, MS <: StaticArrays.StaticArray}
-    a_exp::MS # matrix of size s×s
-    b_exp::VS # vector of length s
-    c_exp::VS # vector of length s
-    a_imp::MS # matrix of size s×s
-    b_imp::VS # vector of length s
-    c_imp::VS # vector of length s
+struct IMEXTableau{AE, BE, CE, AI, BI, CI}
+    a_exp::AE # matrix of size s×s
+    b_exp::BE # vector of length s
+    c_exp::CE # vector of length s
+    a_imp::AI # matrix of size s×s
+    b_imp::BI # vector of length s
+    c_imp::CI # vector of length s
+    function IMEXTableau(a_exp::SArray, b_exp::SArray, c_exp::SArray, a_imp::SArray, b_imp::SArray, c_imp::SArray)
+        args = (a_exp, b_exp, c_exp, a_imp, b_imp, c_imp)
+        mc_args = map(x -> MaskedCoeffs(x), args)
+        return new{typeof.(mc_args)...}(mc_args...)
+    end
+    function IMEXTableau(
+        a_exp::MaskedCoeffs,
+        b_exp::MaskedCoeffs,
+        c_exp::MaskedCoeffs,
+        a_imp::MaskedCoeffs,
+        b_imp::MaskedCoeffs,
+        c_imp::MaskedCoeffs,
+    )
+        mc_args = (a_exp, b_exp, c_exp, a_imp, b_imp, c_imp)
+        return new{typeof.(mc_args)...}(mc_args...)
+    end
```

Comment:
Can we define a method for `Base.convert` that turns an `StaticArray` into a `MaskedCoeffs` (and maybe rename the latter to `SparseCoeffs`)? Then, we can just change the type restriction on `a_exp`, `a_imp`, etc. from `StaticArray` to `MaskedCoeffs`/`SparseCoeffs` and let automatic conversion take care of the rest, instead of needing to add all of these extra constructors.

Reply:
True. I'm not crazy about automatic conversions, but I guess this will only happen during init anyway. I can add these.